### PR TITLE
Fix user service

### PR DESCRIPTION
**Pull request by sad7898**
- Add phone field to `getPersonalInfo`
- Update API docs for `UserService`
- Handle additional error for `uploadImage`

### CODE REVIEW

File: src/user/user.controller.ts [hunk 1]
```diff
@@ -122,24 +130,26 @@ export class UserController {
...
   @ApiUnauthorizedResponse({ description: 'Unauthorized' })
   @UseInterceptors(FileInterceptor('file', { limits: { fileSize: 20000000 } }))
-  @Post('/profile/:uid/image')
+  @Post('profile/image')
   @ApiOkResponse({ type: UploadProfileResponse })
-  @ApiBadRequestResponse({ description: 'Image too large OR Invalid image type' })
+  @ApiBadRequestResponse({ description: 'Image too large' })
+  @ApiUnsupportedMediaTypeResponse({ description: 'Invalid image type' })
   async uploadProfilePicture(@UploadedFile() file: Express.Multer.File, @Request() req): Promise<UploadProfileResponse> {
     const uid = req.user.uid;
```
Comment (tum210544) [inline src/user/user.controller.ts]: no need for this line

File: src/user/user.controller.ts [hunk 2]
```diff
@@ -122,24 +130,26 @@ export class UserController {
...
     const imageUrl = await this.userService.uploadProfilePicture(uid, file);
```
Comment (tum210544) [inline src/user/user.controller.ts]: change from uid to req.user.uid

File: src/user/user.service.ts [hunk 3]
```diff
@@ -128,8 +122,8 @@ export class UserService {
 
   async uploadProfilePicture(uid: number, image: Express.Multer.File): Promise<UploadProfileResponse> {
     const { uid: foundUid } = await this.findOne({ uid: uid }, { shouldExist: true });
-    if (image.mimetype !== 'image/png' && image.mimetype !== 'image/jpeg') {
-      throw new BadRequestException('Invalid image type');
+    if (!image || (image.mimetype !== 'image/png' && image.mimetype !== 'image/jpeg')) {
```
Comment (tum210544) [inline src/user/user.service.ts]: cannot upload pdf?

File: src/user/user.service.ts [hunk 4]
```diff
@@ -128,8 +122,8 @@ export class UserService {
...
+      throw new UnsupportedMediaTypeException('Invalid image type');
     }
     if (image.size > 20000000) {
```
Comment (tum210544) [inline src/user/user.service.ts:129]: dont forget to remind frontend about this constraint na

File: src/user/user.service.ts (hunk 3 above)
Comment (sad7898) [reply src/user/user.service.ts]: I think PDFs are best for documents and large photos, so not ideal for a profile picture.

File: src/user/user.controller.ts (hunk 2 above)
Comment (sad7898) [reply src/user/user.controller.ts]: I'm not sure what you mean. @tum210544

File: src/user/user.controller.ts (hunk 1 above)
Comment (sad7898) [reply src/user/user.controller.ts]: Did you mean line 152? Why not?

File: src/user/user.service.ts (hunk 4 above)
Comment (sad7898) [reply src/user/user.service.ts:129]: okk

File: src/user/user.service.ts (hunk 3 above)
Comment (tum210544) [reply src/user/user.service.ts]: Hmmm I think u should discuss with @pointmekin

**Review by tum210544 — APPROVED** [review]

**Pull request merged**